### ISSUE TITLE
Not able to pass propriety from blade to class

### ISSUE THREAD

**Issue by necroob** [rappasoft/laravel-livewire-tables#693]
I have a blade file with 

`<livewire:dashboard.admin.users.user-orders user_id="{{ $user->id }}"/>`

And in the class

```
class UserOrders extends DataTableComponent
{

    public string $user_id;
    
    public function builder(): Builder
    {
        return Order::query()->where('user_id', $this->user_id);
    }
}
```

It returns this message no matter where I try to access $this->user_id;

`Typed property App\Http\Livewire\Dashboard\Admin\Users\UserOrders::$user_id must not be accessed before initialization`

I tried to use mount() as per the example [here](https://rappasoft.com/docs/laravel-livewire-tables/v2/start/rendering#content-passing-properties), same result.

**Comment by maurocordioli** [comment]
I have similar problems...
and adding some  Log::debug("builder") messages to the builder function it seems to me that
1. the builder function is called multiple times, one per column you define, and some , the leading one's, are called BEFORE the actual "mounting" of the parameters....
as a test **only for the purpose of debugging **  in the builder function  you  could check the property passed by the blade and when is not set return a dummy Builder result lilke <your_orm_class>::query()      
so as eventually the builder is called after the actual mount of the property ... it works but is not a solution.
Maybe we are messing up something else.

**Comment by necroob** [comment]
It doesn't work on my end when trying to do that in the builder. 
It reads the correct value in the mount(), but never reaches the builder.

```
    public string $user_id;

    public function mount(string $user_id): void {
        $this->user_id = $user_id;

        dd( $this->user_id);
    }
```

This displays the correct value. 

However in the builder when I try to use `$this->user_id` I get the error

`Typed property App\Http\Livewire\Dashboard\Admin\Users\UserOrders::$user_id must not be accessed before initialization
`

I don't know how to fix this, since I need to pass that value in the query builder.

**Comment by necroob** [comment]
I ended up doing it like this. 

In my controller:
```
    public function viewUser(User $user)
    {
        return view('admin.users.view', [
            'user' => $user
        ]);
    }
```

Route:
`Route::get('admin/user/{user:id}', [AdminController::class, 'viewUser'])->name('admin-view-user');
`
Livewire Component:
```
    public function builder(): Builder
    {
        $user = request('user');

        return Order::query()->where('user_id', $user->id);
    }
```

I'm not passing the parameter via the livewire component in the blade file.

**Comment by necroob** [comment]
Sooooo... back with another update
I used the suggestion from @maurocordioli 

`<livewire:dashboard.admin.users.user-orders user_id="{{ $user->id }}"/>`

Livewire

```
    public $user_id;

    public function mount(string $user_id): void {
        $this->user_id = $user_id;
    }

    public function builder(): Builder
    {
        if(!empty($this->user_id)) {
            return Order::query()->where('user_id', $this->user_id); 
        }

        return Order::query();
    }
```

Clearly, it's not the ideal way, but it works until we get a more robust solution.

**Comment by falconeri** [comment]
here what i did for now

```
public $campaign;

public function mount(Campaign $campaign)
{
    $this->campaign = $campaign;
}

public function getRows()
{
    $builder = $this->baseQuery();
    return $this->executeQuery($builder->whereCampaignId($this->campaign->id));
}
```

**Comment by maurocordioli** [comment]
@necroob maybe, if you need e temporary workaround the line
> return Order::query();
should be something like "return Order::query()->where('user_id',0);"
(if no order has user_id==0 so it never returns something unintended if future behavior of the component changes ... ) 
the @falconeri suggestion of overriding getRows is even better as a workaround.

**Comment by necroob** [comment]
@maurocordioli Thanks for the suggestion. Yes, that will prevent some unwanted behavior. 
I also tested @falconeri solution, it works, I wasn't sure which one is a better workaround.

**Comment by rappasoft** [comment]
Fixed in v2.4
